src/swupd/manifest.go

```diff
@@ -226,3 +226,113 @@ func (m *Manifest) ReadManifestFromFile(f string) error {
 	// return err so the deferred close can modify it
 	return err
 }
+
+// getManifestTimeStamp returns the unix timestamp of m.Header.TimeStamp
+func (m *Manifest) getManifestTimeStamp() int64 {
+	return m.Header.TimeStamp.Unix()
+}
+
+// writeManifestFileHeader writes the header of a manifest to the file
+func writeManifestFileHeader(m *Manifest, w *bufio.Writer) error {
+	var err error
+	if err = m.CheckHeaderPopulated(); err != nil {
+		return err
+	}
+
+	if _, err = w.WriteString(fmt.Sprintf("MANIFEST\t%v\n", m.Header.Format)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("version:\t%v\n", m.Header.Version)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("previous:\t%v\n", m.Header.Previous)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("filecount:\t%v\n", m.Header.FileCount)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("timestamp:\t%v\n", m.getManifestTimeStamp())); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("contentsize:\t%v\n", m.Header.ContentSize)); err != nil {
+		return err
+	}
+
+	for _, include := range m.Header.Includes {
+		if _, err = w.WriteString(fmt.Sprintf("includes:\t%v\n", include.Name)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// writeManifestFileSep writes a blank line to the file
+func writeManifestFileSep(w *bufio.Writer) error {
+	_, err := w.WriteString("\n")
+	return err
+}
+
+// writeManifestFileEntry writes the file entry line to the file
+func writeManifestFileEntry(file *File, w *bufio.Writer) error {
+	var flags string
+	var err error
+	if flags, err = file.getFlagString(); err != nil {
+		return err
+	}
+
+	line := fmt.Sprintf("%v\t%v\t%v\t%v\n",
+		flags,
+		file.getHashString(),
```

Comment:
At least my intention here was that you could simply say file.Hash here, and you shouldn't need the getHashString method at all, the String method on hashval should take care of it.

In particular I could imagine in the future storing the hash as a 32 byte binary object, rather than a 64 (+1 for terminating NUL) string of hex characters to reduce space.